### PR TITLE
Improved logging

### PR DESCRIPTION
**Pull request by DaJansenGit**
Some improvements of the current loggin:
* [X] Reduce doubled log messsages. This was due to doubled logger handlers which are now cleaned 
* [X] colored logging messages based on level of the log message
* [X] improved some log messages for TEASER

Still some work to do for better error handlilng.

### CODE REVIEW

File: bim2sim/project.py [hunk 1]
```diff
@@ -491,7 +495,8 @@ def run(self, interactive=False, cleanup=True):
 
         if not self._user_logger_set:
             logger.info("Set user logger to default Stream. "
-                        "Call project.set_user_logger_stream() prior to project.run() to change this.")
+                        "Call project.set_user_logger_stream() prior to "
```
Comment (veronikarichter) [inline bim2sim/project.py:498]: set_user_logger_stream() doesn't exist. 

How do I change the logging level when running an example?

**Review by veronikarichter — CHANGES_REQUESTED** [review]

File: bim2sim/task/bps/bind_tz.py [hunk 2]
```diff
@@ -69,40 +74,20 @@ def bind_tz_criteria(self, instances):
                 self.bounded_tz.append(inst)
 
     @classmethod
-    def group_thermal_zones_by_all_criteria(cls, instances):
+    def group_thermal_zones_by_is_external(cls, instances):
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/task/bps/bind_tz.py:77]: not in any way, just wanted to improve the decisions. I have to addmit it not really fits into this PR but is also related to command line output, so please be indulgent :)

File: bim2sim/task/bps/bind_tz.py [hunk 3]
```diff
@@ -162,6 +139,34 @@ def group_by_usage(cls, thermal_zones: list) -> dict:
         cls.discard_1_element_groups(grouped_tz)
         return grouped_tz
 
+    @classmethod
+    def group_thermal_zones_by_use_all_criteria(cls, instances):
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/task/bps/bind_tz.py:143]: see answer above

File: bim2sim/task/bps/bind_tz.py [hunk 4]
```diff
@@ -52,11 +54,14 @@ def bind_tz_criteria(self, instances):
         for k, func in dict(
                 inspect.getmembers(self, predicate=inspect.ismethod)).items():
             if k.startswith('group_thermal_zones_'):
-                criteria_functions[k.replace('group_thermal_zones_', '')] = func
+                criteria_functions[
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/task/bps/bind_tz.py:57]: see answer above

File: bim2sim/project.py (hunk 1 above)
Comment (DaJansenGit) [reply bim2sim/project.py:498]: > set_user_logger_stream() doesn't exist.

I fixed this with latest commit.

> How do I change the logging level when running an example?

If you want to put the user log to another place so you don't see everything in console you could do the following:
### change user log handler:
* This is described in the [documentation of the logging concept ](https://ebc.pages.rwth-aachen.de/EBC_all/github_ci/bim2sim/development/docs/concepts/logging.html)
* E.g. you could add these statements before running `project.run()`

```
    from logging import FileHandler

    my_handler = FileHandler("my_log_file.log")

    project.set_user_logging_handler(my_handler)
```
* Then you won't have any log messages in the console from the user logging as they will be writen to the my_lof_file.log file in your CWD.

### If you just want to change the threshold level from which on you get logs you can do the following:
```
    general_logger = logging.getLogger('bim2sim')

    general_logger.setLevel(logging.WARNING)
```
This way you will only receiver Warnings and Errors and Critials, see [logging module docs](https://docs.python.org/3/library/logging.html#levels) as well

**Review by veronikarichter — APPROVED** [review]

**Pull request merged**